scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -760,10 +792,61 @@ class FossId internal constructor(
         scanId: String
     ): ScanResult {
         // TODO: Maybe get issues from FossID (see has_failed_scan_files, get_failed_files and maybe get_scan_log).
+
+        // TODO: Deprecation: Remove the pending files in issues. This is a breaking change.
         val issues = rawResults.listPendingFiles.mapTo(mutableListOf()) {
             Issue(source = name, message = "Pending identification for '$it'.", severity = Severity.HINT)
         }
 
+        val snippetFindings = mutableSetOf<SnippetFinding>()
+        val fakeLocation = TextLocation(".", TextLocation.UNKNOWN_LINE)
+        snippetFindings += rawResults.listSnippets.flatMap { (file, rawSnippets) ->
+            val snippets = rawSnippets.map {
+                val license = it.artifactLicense?.let {
+                    runCatching {
+                        LicenseFinding.createAndMap(
+                            it,
+                            fakeLocation,
+                            detectedLicenseMapping = scannerConfig.detectedLicenseMapping
+                        ).license
+                    }.onFailure { spdxException ->
+                        issues += FossId.createAndLogIssue(
+                            source = "FossId",
+                            message = "Failed to parse license '$it' as an SPDX expression:" +
+                                    " ${spdxException.collectMessages()}"
+                        )
+                    }.getOrNull()
+                } ?: SpdxConstants.NOASSERTION.toSpdx()
+
+                // FossID does not return the hash of the remote artifact: it returns instead, in the property
+                // "match_file_id", the MD5 hash of the matched file IN the remote artifact.
+                val snippetProvenance = it.url?.let { url ->
+                    ArtifactProvenance(RemoteArtifact(url, Hash.NONE))
+                } ?: UnknownProvenance
+                val purlType = it.url?.let { url -> urlToPackageType(url, issues)?.toString() } ?: "generic"
+
+                // TODO: FossId doesn't return the line numbers of the match, only the character range. One must use
+                //  another call "getMatchedLine" to retrieve the matched line numbers. Unfortunately, this is a call
+                //  per snippet which is too expensive. When it is available for a batch of snippets, it can be used
+                //  here.
```

Comment:
Wrong indentation: Align to the word "FossID" in the first line (and consistently spell it like "FossID", not "FossId", in free text).